Source/Core/VideoCommon/Assets/MeshAsset.cpp

```diff
@@ -0,0 +1,597 @@
+{
+  const tinygltf::Accessor& accessor = model.accessors[accessor_index];
+
+  const int component_count = tinygltf::GetNumComponentsInType(accessor.type);
+  if (component_count == -1)
+  {
+    ERROR_LOG_FMT(VIDEO, "Failed to copy buffer data from primitive, component count was invalid");
+    return false;
+  }
+
+  const int component_size =
+      tinygltf::GetComponentSizeInBytes(static_cast<uint32_t>(accessor.componentType));
+  if (component_size == -1)
+  {
+    ERROR_LOG_FMT(VIDEO, "Failed to copy buffer data from primitive, component size was invalid");
+    return false;
+  }
+
+  const tinygltf::BufferView& buffer_view = model.bufferViews[accessor.bufferView];
+  const tinygltf::Buffer& buffer = model.buffers[buffer_view.buffer];
+
+  if (buffer_view.byteStride == 0)
+  {
+    // Data is tightly packed
+    const auto data = &buffer.data[accessor.byteOffset + buffer_view.byteOffset];
+    for (std::size_t i = 0; i < accessor.count; i++)
+    {
+      const std::size_t vertex_data_offset = i * chunk->vertex_stride + *outbound_offset;
+      memcpy(&chunk->vertex_data[vertex_data_offset], &data[i * component_size * component_count],
+             component_size * component_count);
+    }
+  }
+  else
+  {
+    // Data is interleaved
+    const auto data = &buffer.data[accessor.byteOffset + buffer_view.byteOffset];
+    for (std::size_t i = 0; i < accessor.count; i++)
+    {
+      const std::size_t vertex_data_offset = i * chunk->vertex_stride + *outbound_offset;
+      const std::size_t gltf_data_offset = i * buffer_view.byteStride;
+
+      memcpy(&chunk->vertex_data[vertex_data_offset], &data[gltf_data_offset],
+             component_size * component_count);
+    }
+  }
+
+  *outbound_offset += component_size * component_count;
+
+  return true;
+}
+
+bool ReadGLTFMesh(std::string_view mesh_file, const tinygltf::Model& model,
+                  const tinygltf::Mesh& mesh, const Common::Matrix44& mat, MeshData* data)
+{
+  for (std::size_t primitive_index = 0; primitive_index < mesh.primitives.size(); ++primitive_index)
+  {
+    MeshDataChunk chunk;
+    chunk.transform = mat;
+    const tinygltf::Primitive& primitive = mesh.primitives[primitive_index];
+    if (primitive.indices == -1)
+    {
+      ERROR_LOG_FMT(VIDEO, "Mesh '{}' is expected to have indices but doesn't have any", mesh_file);
+      return false;
+    }
+    chunk.material_name = model.materials[primitive.material].name;
+    const tinygltf::Accessor& index_accessor = model.accessors[primitive.indices];
+    const tinygltf::BufferView& index_buffer_view = model.bufferViews[index_accessor.bufferView];
+    const tinygltf::Buffer& index_buffer = model.buffers[index_buffer_view.buffer];
+    const int index_stride = index_accessor.ByteStride(index_buffer_view);
+    if (index_stride == -1)
+    {
+      ERROR_LOG_FMT(VIDEO, "Mesh '{}' has invalid stride", mesh_file);
+      return false;
+    }
+    chunk.indices = std::make_unique<u16[]>(index_accessor.count);
```

Comment:
```suggestion
    chunk.indices = std::unique_ptr<u16[]>(new u16[index_accessor.count]);
```

if all the elements are guaranteed to be initialized below, since the previous code zeroes out everything. Ideally this would be std::make_unique_for_overwrite, but it's c++23 and up only.

Reply:
Thanks, C++23 can't come soon enough..